cfgov/v1/models/filterable_list_mixins.py

```diff
@@ -41,29 +42,36 @@ def get_search_class():
         return FilterablePagesDocumentSearch
 
     def get_filterable_list_wagtail_block(self):
+        """Find a FilterableList StreamField block in a content field."""
         return next(
             (b for b in self.content if b.block_type == "filter_controls"),
             None,
         )
 
-    def get_filterable_root(self):
-        filterable_list_block = self.get_filterable_list_wagtail_block()
-        if filterable_list_block is None:
-            return "/"
-
-        if filterable_list_block.value["filter_children"]:
-            return self.get_url()
+    def get_filterable_search(self):
+        # By default, filterable pages only search their direct children.
+        # But this can be overriden by a setting on a FilterableList block
+        # added to the page's content StreamField.
+        if filter_block := self.get_filterable_list_wagtail_block():
+            children_only = filter_block.value.get("filter_children", True)
+        else:
+            children_only = True
 
-        return "/"
+        # If searching globally, use the root page of this page's Wagtail
+        # Site. If the page doesn't live under a Site (for example, it is
+        # in the Trash), use the default Site.
+        if not children_only:
+            site = self.get_site()
 
-    def get_filterable_search(self):
-        """Return a FilterablePagesDocumentSearch object"""
-        site = self.get_site()
+            if not site:
+                site = Site.objects.get(is_default_site=True)
```

Comment:
It's kind of a shame that `get_site()` doesn't provide an optional way to fetch the default site.